devices/snesclassicfactory.cpp

```diff
@@ -191,6 +228,7 @@ void SNESClassicFactory::aliveCheck()
     {
         sDebug() << "Closing the device, Canoe not running anymore";
         device->close();
+        socket->close();
```

Comment:
You don't really need to close the connection, you can still do other requests or the device will get cleaned and the socket closed